src/main/java/io/tackle/operator/KeycloakController.java

```diff
@@ -32,15 +36,33 @@
 
     @Inject
     KubernetesClient kubernetesClient;
+    @Inject
+    PostgreSQLDeployer postgreSQLDeployer;
 
     @Override
     public UpdateControl<Keycloak> createOrUpdateResource(Keycloak keycloak, Context<Keycloak> context) {
         String namespace = keycloak.getMetadata().getNamespace();
         // Keycloak is unique, no need for suffixes in the name
         String name = metadataName(keycloak);
 
+        MixedOperation<Keycloak, KubernetesResourceList<Keycloak>, Resource<Keycloak>> keycloakClient = kubernetesClient.customResources(Keycloak.class);
+        MixedOperation<Tackle, KubernetesResourceList<Tackle>, Resource<Tackle>> tackleClient = kubernetesClient.customResources(Tackle.class);
+        if (tackleClient.inNamespace(namespace).list().getItems().isEmpty()) {
+            log.errorf("Standalone '%s' Keycloak CR isn't allowed: create a Tackle CR to instantiate Tackle application", name);
+            keycloakClient.delete(keycloak);
+            return UpdateControl.noUpdate();
+        } else if (keycloakClient.inNamespace(namespace).list().getItems().size() > 1) {
```

Comment:
Shouldn't it be checking that not more than 1 keycloak CR is created for a given Tackle CR ? In that case the user could create several tackles in the same namespace.

Reply:
:+1: 
Implemented the "Multiple Tackle CRs in a single namespace" approach.